src/security/BlindPeerConnector.cc

```diff
@@ -70,13 +70,13 @@ Security::BlindPeerConnector::noteNegotiationDone(ErrorState *error)
 
     if (error) {
         debugs(83, 5, "error=" << (void*)error);
-        // XXX: forward.cc calls peerConnectSucceeded() after an OK TCP connect but
-        // we call peerConnectFailed() if SSL failed afterwards. Is that OK?
-        // It is not clear whether we should call peerConnectSucceeded/Failed()
+        // XXX: FwdState calls NoteOutgoingConnectionSuccess() after an OK TCP connect, but
+        // we call noteFailure() if SSL failed afterwards. Is that OK?
```

Comment:
To answer the question: this is not okay. noteSuccess() is our callers responsibility if/when their transaction completes (i.e. the client gets what they want via this peer).

I am not sure whether to request just fixing this XXX message, or the bug behind it. You have fixed similar bugs in this PR and it should be easy to just drop noteSuccess() from this class.

Reply:
This is no easy fix here, some fixes will have non-trivial effects on deployed Squids, and there is currently no consensus whether your answer is correct. The current official/PR code does not work the way you describe. We have an in-progress PR that will address this XXX. Let's postpone this discussion until then. This old XXX and all those related problems are out of this PR scope.

Reply:
Ack, FTR I described the ideal (many bugs fail to meet up to it).

Since you have a followup already underway I will not insist.